plugin-base/src/main/java/io/everytrade/server/plugin/impl/everytrade/BittrexConnector.java

```diff
@@ -97,100 +100,184 @@ public DownloadResult getTransactions(String stateStr) {
+        String startItemId = null;
+        String endItemId = null;
+
+        if (startStateItemId != null && endStateItemId == null) {
+            startItemId = startStateItemId;
         }
 
-        if (!withdrawals.isEmpty()) {
-            downloadState.setLastDepositId(withdrawals.get(withdrawals.size() -1).getId());
+        if (endStateItemId != null) {
+            endItemId = endStateItemId;
+        }
+
+        int sentRequests = 0;
+        while (sentRequests < MAX_REQUEST_COUNT) {
+            ++sentRequests;
+            final List<BittrexWithdrawalHistory> withdrawalBlock = new ArrayList<>();
+
+            try {
+                withdrawalBlock.addAll(accountService.getBittrexWithdrawalsClosed(
+                    null, endItemId, startItemId, DEPOSIT_WITHDRAWAL_PAGE_SIZE
+                ));
+            } catch (IOException e) {
+                throw new IllegalStateException("User withdrawal history download failed.", e);
+            }
+            // first round
+            if (endItemId == null && !withdrawalBlock.isEmpty()) {
+                String id = withdrawalBlock.get(0).getId(); // add one
+                downloadState.setStartWithdrawalId(id);
+            }
+
+            if (withdrawalBlock.isEmpty() || withdrawalBlock.size() < DEPOSIT_WITHDRAWAL_PAGE_SIZE) {
+                downloadState.setEndWithdrawalId(null);
+                break;
+            }
+            endItemId = withdrawalBlock.get(withdrawalBlock.size() - 1).getId();
+            downloadState.setEndDepositId(endItemId);
+            withdrawals.addAll(withdrawalBlock);
         }
         return withdrawals;
     }
 
+
     @Data
     @NoArgsConstructor
     @AllArgsConstructor
     @FieldDefaults(level = PRIVATE)
     private static class DownloadState {
         private static final String SEPARATOR = "=";
 
-        String lastTxId;
-        String lastDepositId;
-        String lastWithdrawalId;
+        String startTxTime;
+        String endTxTime;
+        String startDepositId;
+        String endDepositId;
+        String startWithdrawalId;
+        String endWithdrawalId;
 
         public static DownloadState deserialize(String state) {
             if (isEmpty(state)) {
                 return new DownloadState();
             }
-            var strA = state.split(SEPARATOR);
+            var stringState = state.split(SEPARATOR);
+
             return new DownloadState(
-                strA[0],
-                strA.length > 1 ? emptyToNull(strA[1]) : null,
-                strA.length > 2 ? emptyToNull(strA[2]) : null
+                stringState[0].equals(" ") ? null : stringState[0],
+                stringState[1].equals(" ") ? null : stringState[1],
+                stringState[2].equals(" ") ? null : stringState[2],
+                stringState[3].equals(" ") ? null : stringState[3],
+                stringState[4].equals(" ") ? null : stringState[4],
+                stringState[5].equals(" ") ? null : stringState[5]
```

Comment:
you should make check for length too. So far we have only 3 values in DB (and not everytime, some old turnoff connectors have only 1 value and no `=` char separator), so`stringState[1]`  will throw exception. Make this check for all values.